src/Application/Lockers/Commands/UpdateLocker.cs

```diff
@@ -1,15 +1,83 @@
+using Application.Common.Exceptions;
+using Application.Common.Interfaces;
 using Application.Common.Models.Dtos.Physical;
+using AutoMapper;
+using Domain.Entities.Physical;
+using Domain.Exceptions;
+using FluentValidation;
 using MediatR;
+using Microsoft.EntityFrameworkCore;
 
 namespace Application.Lockers.Commands;
 
 public class UpdateLocker
 {
+    public class Validator : AbstractValidator<Command>
+    {
+        public Validator()
+        {
+            RuleLevelCascadeMode = CascadeMode.Stop;
+            
+            RuleFor(x => x.Name)
+                .NotEmpty()
+                .MaximumLength(64).WithMessage("Locker's name cannot exceed 64 characters.");
+            
+            RuleFor(x => x.Description)
+                .MaximumLength(256).WithMessage("Locker's description cannot exceed 256 characters.");
+            
+            RuleFor(x => x.Capacity)
+                .GreaterThan(0).WithMessage("Locker's capacity cannot be less than 1");
+        }
+    }
     public record Command : IRequest<LockerDto>
     {
         public Guid LockerId { get; init; }
         public string Name { get; init; } = null!;
         public string? Description { get; init; }
         public int Capacity { get; init; }
     }
+    
+    public class CommandHandler : IRequestHandler<Command, LockerDto>
+    {
+        private readonly IApplicationDbContext _context;
+        private readonly IMapper _mapper;
+
+        public CommandHandler(IApplicationDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task<LockerDto> Handle(Command request, CancellationToken cancellationToken)
+        {
+            var locker = await _context.Lockers.FirstOrDefaultAsync(
+                x => x.Id.Equals(request.LockerId), cancellationToken);
+
+            var duplicateLocker = await _context.Lockers.FirstOrDefaultAsync(
+                x => x.Name.Equals(request.Name), cancellationToken);
+            
+            if (locker is null)
+            {
```

Comment:
this check should be before duplicateLocker

Reply:
what?